File: Recursion_1/python/power.py
# Problem : Power
def power(x,n):
	if n == 0:
		return 1
	smallAns = power(x,n-1)
	return smallAns;

# Main
from sys import setrecursionlimit
setrecursionlimit(1100)
x,n = list(int(i) for i in input().strip().split(' '))
print(power(x,n))

File: Recursion_1/java/power.java
package Recursion_1.java;
class Power {
    public static void main(String[] args) {
        System.out.println();
        power(3, 2);
    }
    public static int power(int x, int n) {
        if (x == 0 && n == 0) {
            return 1;
        }
        if (x == 0) {
            return 0;
        }
        if (n == 0) {
            return 1;
        }
        int smallAns = power(x, n / 2);
        if (n % 2 == 0) {
            return smallAns * smallAns;
        } else {
            return x * smallAns * smallAns;
        }
    }
}